firebase-app-distribution/src/main/java/com/google/firebase/appdistribution/TesterSignInClient.java

```diff
@@ -75,22 +76,24 @@
     this.signInStorage = signInStorage;
     this.lifecycleNotifier = lifecycleNotifier;
 
+    lifecycleNotifier.addOnActivityCreatedListener(this::onActivityCreated);
     lifecycleNotifier.addOnActivityStartedListener(this::onActivityStarted);
     lifecycleNotifier.addOnActivityDestroyedListener(this::onActivityDestroyed);
   }
 
-  private void onActivityDestroyed(Activity activity) {
-    this.dismissAlertDialog();
-  }
-
   @VisibleForTesting
```

Comment:
Out of curiosity why can't this happen onActivityStarted? Also would be great if you could add a comment explaining why we have different things happening onActivityCreated and onActivityStarted

Reply:
The flow I noticed was SignInResultActivity Created -> MainActivityCreated -> MainActivityStarted.

I am not sure why SignInResultActivity was not started before MainActivity. Any theories as to why that might happen? 

This logic lived in onCreate originally, so I assumed my implementation was buggy, and we had some reason for putting in onCreate in the original implementation

Reply:
nvm, I see why. We call finish() in the onCreate of SignInResultActivity. 

I'll add a comment :)